app/GRAPHQL/components/GqlForm/GQLForm.tsx

```diff
@@ -0,0 +1,254 @@
+'use client';
+
+import { createQuery } from '@/lib/actions/form.actions';
+import { useEffect, useRef, useState } from 'react';
+import SubmitButton from '../buttons/SubmitButton';
+import styles from './gqlForm.module.css';
+import { useFormState } from 'react-dom';
+import PrettifyButton from '../buttons/PrettifyButton';
+import SDLButton from '../SDLButton/SDLButton';
+import gqlPrettier from 'graphql-prettier';
+import EndpointURL from '../endpointUrl/EndpointURL';
+import EndpointSDL from '../endpointSDL/EndpointSDL';
+import QuerySection from '../querySection/QuerySection';
+import VariablesSection from '../variables/VariablesSection';
+import ExplorerButton from '../buttons/ExplorerButton';
+import Headers from '../headers/Headers';
+import SchemaDocumentation from '../schemaDocumentation/SchemaDocumentation';
+import { ResultBlock } from '@/app/components/REST/components/ResultBlock';
+import {
+  createGqlFormSchema,
+  createPrettifySchema,
+} from '@/lib/formValidationSchema/validationSchema';
+import * as Yup from 'yup';
+import { usePathname, useSearchParams } from 'next/navigation';
+import { decodeBase64, saveRequestToLS } from '@/app/[...rest]/utils';
+import { useTranslations } from 'next-intl';
+
+const errMessages = [
+  'Invalid URL format.',
+  'Неправильный URL формат.',
+  'Неправильный GraphQL запрос.',
+  'Invalid GraphQL query.',
+  'Invalid JSON format.',
+  'Неправильный JSON формат.',
+];
+
+const errMessagesUrl = [
+  'Either endpointURL or endpointSDL is required.',
+  'Требуется адрес URL, либо адрес SDL.',
+];
+
+const GQLForm = () => {
+  const t = useTranslations('apiClient');
+  const gqlFormSchema = createGqlFormSchema(t);
+  const prettifySchema = createPrettifySchema(t);
+
+  const ref = useRef<HTMLFormElement>(null);
+  const [data, action] = useFormState(createQuery, {
+    title: '',
+    status: null,
+    message: '',
+  });
+  const pathname = usePathname();
+  const search = useSearchParams().toString();
+  const [endpointURL, setEndpointURL] = useState<string>(() => {
+    const pathArray = pathname.split('/');
+    const url = pathArray[2];
+    const decodedUrl = url ? decodeURIComponent(decodeBase64(url)) : '';
+    return decodedUrl;
+  });
+  const [endpointSDL, setEndpointSDL] = useState<string>('');
+  const [variablesArea, setVariablesArea] = useState<string>(() => {
+    const pathArray = pathname.split('/');
+    const query = pathArray[3];
+    const decodedVariables = query
+      ? JSON.parse(decodeURIComponent(decodeBase64(query))).variables
+      : '';
+    return decodedVariables;
+  });
+
+  const [queryArea, setQueryArea] = useState<string>(() => {
+    const pathArray = pathname.split('/');
+    const query = pathArray[3];
+    const decodedQuery = query
+      ? JSON.parse(decodeURIComponent(decodeBase64(query))).query
+      : '';
+    return decodedQuery;
+  });
+  const [open, setOpen] = useState<boolean>(false);
+  const [show, setShow] = useState<boolean>(false);
```

Comment:
```suggestion
  const [isShown, setIsShow] = useState<boolean>(false);
```
It's better to use predicates for the boolean constants naiming.

Reply:
Done ${5cff4a636940ae3feb595dbe4c319a1855dfe1c0}